From 3f0b34ab5f0889c49d3484b07e0045b2ef445f70 Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Sat, 8 May 2021 11:30:32 +0200
Subject: [PATCH] ath79: various fixes

* free up additional 1MB rootfs space
* make the target use bridge-vlan
* use the single eth port as wan

Signed-off-by: John Crispin <john@phrozen.org>
---
 target/linux/ath79/dts/ar9344_tplink_cpe.dtsi       | 13 ++++++++++---
 .../ath79/generic/base-files/etc/board.d/02_network |  3 ++-
 tools/firmware-utils/src/tplink-safeloader.c        |  3 ++-
 3 files changed, 14 insertions(+), 5 deletions(-)

diff --git a/target/linux/ath79/dts/ar9344_tplink_cpe.dtsi b/target/linux/ath79/dts/ar9344_tplink_cpe.dtsi
index fd6aa0f08b..2b663a938e 100644
--- a/target/linux/ath79/dts/ar9344_tplink_cpe.dtsi
+++ b/target/linux/ath79/dts/ar9344_tplink_cpe.dtsi
@@ -60,9 +60,16 @@
 			partition@40000 {
 				label = "firmware";
 				reg = <0x040000 0x780000>;
-				compatible = "openwrt,uimage", "denx,uimage";
-				openwrt,ih-magic = <IH_MAGIC_OKLI>;
-				openwrt,offset = <0x3000>;
+			};
+
+			partition@040000 {
+				label = "kernel";
+				reg = <0x040000 0x210000>;
+			};
+
+			partition@340000 {
+				label = "rootfs";
+				reg = <0x250000 0x560000>;
 			};
 
 			partition@7c0000 {
diff --git a/target/linux/ath79/generic/base-files/etc/board.d/02_network b/target/linux/ath79/generic/base-files/etc/board.d/02_network
index d70432ad2b..b0b152ba74 100755
--- a/target/linux/ath79/generic/base-files/etc/board.d/02_network
+++ b/target/linux/ath79/generic/base-files/etc/board.d/02_network
@@ -95,7 +95,7 @@ ath79_setup_interfaces()
 	ubnt,unifi|\
 	wd,mynet-wifi-rangeextender|\
 	winchannel,wb2000)
-		ucidef_set_interface_lan "eth0"
+		ucidef_set_interface_wan "eth0"
 		;;
 	airtight,c-75)
 		ucidef_add_switch "switch0" \
@@ -675,6 +675,7 @@ ath79_setup_macs()
 
 board_config_update
 board=$(board_name)
+ucidef_set_bridge_device bridge
 ath79_setup_interfaces $board
 ath79_setup_macs $board
 board_config_flush
diff --git a/tools/firmware-utils/src/tplink-safeloader.c b/tools/firmware-utils/src/tplink-safeloader.c
index c519a6d367..a0390c3797 100644
--- a/tools/firmware-utils/src/tplink-safeloader.c
+++ b/tools/firmware-utils/src/tplink-safeloader.c
@@ -439,7 +439,8 @@ static struct device_info boards[] = {
 			{"default-mac", 0x30000, 0x00020},
 			{"product-info", 0x31100, 0x00100},
 			{"signature", 0x32000, 0x00400},
-			{"firmware", 0x40000, 0x770000},
+			{"os-image", 0x40000, 0x210000},
+			{"file-system", 0x250000, 0x560000},
 			{"soft-version", 0x7b0000, 0x00100},
 			{"support-list", 0x7b1000, 0x00400},
 			{"user-config", 0x7c0000, 0x10000},
-- 
2.25.1

